dagshub/data_engine/datasets.py

```diff
@@ -41,7 +40,7 @@ def _from_gql_result(repo: str, dataset_result: "DatasetResult") -> "Datasource"
     ds = Datasource(DatasourceState.from_gql_result(repo, dataset_result.datasource))
 
     query_dict = json.loads(dataset_result.datasetQuery)
-    if "query" in query_dict:
-        ds._query = DatasourceQuery.deserialize(query_dict["query"])
+
+    Datasource.deserialize_gql_result(ds, query_dict)
```

Comment:
```suggestion
    ds.deserialize_gql_result(query_dict)
```

Also probably should make it private by renaming it to `ds._deserialize_gql_result`, so users/IDEs won't think it's part of the public interface.